src/analysis/hofer.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2017 Patrick Massot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot
+-/
+
+import analysis.specific_limits
+
+/-!
+# Hofer's lemma
+
+This is an elementary lemma about complete metric space. It is motivated by
+application to the bubbling-off analysis for holomorphic curves in symplectic topology.
+We are *very* far away from having these applications, but the proof here is a nice
+example of a proof needing to construct a sequence by induction in the middle of the proof.
+
+## References:
+
+* H. Hofer and C. Viterbo, *The weinstein conjecture in the presence of holomorphic spheres*
+-/
+
+open_locale classical topological_space big_operators
+open filter finset
+
+local notation `d` := dist
+
+lemma hofer {X: Type*} [metric_space X] [complete_space X]
```

Comment:
Should this be in a namespace?

Reply:
I expect this file won't be imported anyway. We could even put it in the archive folder if people complain about it.